core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -186,6 +188,66 @@ private[spark] object JettyUtils extends Logging {
     contextHandler
   }
 
+  /** Create a handler for proxying request to Workers and Application Drivers */
+  def createProxyHandler(
+      prefix: String,
+      target: String): ServletContextHandler = {
+    val servlet = new ProxyServlet {
+      override def rewriteTarget(request: HttpServletRequest): String = {
+        val path = request.getRequestURI();
+        if (!path.startsWith(prefix)) return null
+
+        val uri = new StringBuilder(target)
+        if (target.endsWith("/")) uri.setLength(uri.length() - 1)
```

Comment:
Seems like it would be better to normalize `target` and `prefix` before instantiating the servlet, so that they conform to whatever you expect, and avoid these extra checks on every invocation of the method.

e.g., make sure `target` does not end with a slash, and `prefix` does.